internal/idp/providers/ldap/session.go

```diff
@@ -30,15 +32,37 @@ type Session struct {
 func (s *Session) GetAuthURL() string {
 	return s.loginUrl
 }
-func (s *Session) FetchUser(_ context.Context) (idp.User, error) {
+func (s *Session) FetchUser(_ context.Context) (_ idp.User, err error) {
 	var user *ldap.Entry
-	var err error
 	for _, server := range s.Provider.servers {
+		conn, err := ldap.DialURL(server, ldap.DialWithDialer(&net.Dialer{Timeout: s.Provider.timeout}))
+		if err != nil {
+			// Failed DialURL start with next server
+			continue
+		}
+		defer conn.Close()
+
+		u, err := url.Parse(server)
+		if err != nil {
+			// Failed URL parsing end with error
+			return nil, err
+		}
+		if u.Scheme == "ldaps" && s.Provider.startTLS {
+			err = conn.StartTLS(&tls.Config{ServerName: u.Host})
+			if err != nil {
+				// Failed StartTLS connection try with next server
+				continue
+			}
+		}
+		// Bind as the admin to search for user
+		err = conn.Bind(s.Provider.bindDN, s.Provider.bindPassword)
+		if err != nil {
+			// Failed bind try with next server
+			continue
+		}
+
```

Comment:
please outsource that method. that way you're also able to defer correctly.
do we need to handle the parse error differently than the other errors?